src/daos/set_integer.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+ACCEPTABLE_INTEGER_REGEX='[0-9]+'
+
+function set_integer()
+{
+    declare -n foo=$1
+    local not_used=`grep -E "^${ACCEPTABLE_INTEGER_REGEX}$" <<< $2`
+    if [[ $? == 0 ]]; then
```

Comment:
$? doesn't reflect the exit status of "grep". For example, variables "foo" and "$1" are assigned  "$2"  even if  $2 is non a integer value.

Reply:
why not, so we are checking if `$2` is a valid integer value or not, $? should contain the value of recent executed command.
consider this,
```
ACCEPTABLE_INTEGER_REGEX='[0-9]+'

not_used=`grep -E "^${ACCEPTABLE_INTEGER_REGEX}$" <<< 2`
echo $?
```
output: 0

```
ACCEPTABLE_INTEGER_REGEX='[0-9]+'

not_used=`grep -E "^${ACCEPTABLE_INTEGER_REGEX}$" <<< asdasd`
echo $?
```

output: 1

Reply:
It works without "local" keyword,  but doesn't work with "local" keyword.

var1=50
echo before $var1
set_integer var1 "abc"
echo after $var1
-----------------------------
OUTPUT:
before 50
after abc

Reply:
Anyway, as you suggested,  we can fix it later